tests/testthat/test-config-formulas.R

```diff
@@ -0,0 +1,67 @@
+context("config")
+
+input <- list(population = 1000,
+                      procurePeoplePerNet = 1.8,
+                      priceNetStandard = 1.5,
+                      priceNetPBO = 2.5,
+                      priceDelivery = 2.75,
+                      procureBuffer = 7,
+                      priceIRSPerPerson = 2.5)
+
+# setting these variables up to be as similar as possible to Ellie's original code
+# https://github.com/mrc-ide/shiny-malaria-UI/blob/master/server.R#L1148
+# for verification of correctness
+population <- input$population
+procurement_buffer <- (input$procureBuffer + 100) / 100
+cost_per_N0 <- input$priceNetStandard
+cost_per_N1 <- input$priceNetStandard
+cost_per_N2 <- input$priceNetPBO
+price_NET_delivery <- input$priceDelivery *  population *  procurement_buffer
+price_IRS_delivery <- input$priceIRSPerPerson * population
+
+costs_N0 <- cost_per_N0 * (population/input$procurePeoplePerNet) + 0
+costs_N1 <- cost_per_N1 * (population/input$procurePeoplePerNet) + price_NET_delivery
+costs_N2 <- cost_per_N2 * (population/input$procurePeoplePerNet) + price_NET_delivery
+costs_S1 <- cost_per_N0 * (population/input$procurePeoplePerNet) + 0 + 3 * price_IRS_delivery
+costs_N1_S1 <- cost_per_N1 * (population/input$procurePeoplePerNet) + price_NET_delivery + 3 * price_IRS_delivery
+costs_N2_S1 <- cost_per_N2 * (population/input$procurePeoplePerNet) + price_NET_delivery + 3 * price_IRS_delivery
+
+evaluate <- function(formula) {
+  formula <- stringr::str_replace_all(formula, stringr::fixed("{"), "${")
+  formula <- stringr::str_interp(formula, input)
+  eval(parse(text = formula))
```

Comment:
This can be done somewhat more simply (and with a lighter dependency) with glue:

```suggestion
  eval(parse(text = glue::glue(formula, .envir = input)))
```

Reply:
I guessed there might be a better way, cheers